content/articles/2018-04-readxl-1-1-0.Rmd

```diff
@@ -0,0 +1,166 @@
+
+  * [CVE-2017-12110](https://www.talosintelligence.com/vulnerability_reports/TALOS-2017-0462)
+  * [CVE-2017-2896](https://www.talosintelligence.com/vulnerability_reports/TALOS-2017-0403)
+  * [CVE-2017-2897](https://www.talosintelligence.com/vulnerability_reports/TALOS-2017-0404)
+  
+Those have now been addressed upstream and version 1.1.0 of readxl embeds a version of libxls that includes those fixes.
+
+FYI, there are two other CVEs for libxls that have long been resolved in libxls itself, that do not affect readxl, have never affected readxl, but nonetheless still mention readxl in their text. [CVE-2017-12111](https://www.talosintelligence.com/vulnerability_reports/TALOS-2017-0463), [CVE-2017-2919](https://www.talosintelligence.com/vulnerability_reports/TALOS-2017-0426)
+
+The recent maintenance of libxls is a very positive development and has allowed us to close many readxl issues related to crashes or other unsavory behaviour seen when reading specific xls files.
+
+## Is it xls or xlsx?
+
+readxl has 3 main functions for reading Excel files:
+
+  * [`read_xlsx()`](http://readxl.tidyverse.org/reference/read_excel.html) reads input as xlsx
+  * [`read_xls()`](http://readxl.tidyverse.org/reference/read_excel.html) reads input as xls
+  * [`read_excel()`](http://readxl.tidyverse.org/reference/read_excel.html) determines whether the input is xls or xlsx, then calls one of the functions above
+    
+`read_excel()` has gotten better at inferring the file format and this functionality is now exposed, in case others need access. This is implemented in [`excel_format()`](http://readxl.tidyverse.org/reference/excel_format.html), which first calls [`format_from_ext()`](http://readxl.tidyverse.org/reference/excel_format.html) to consult the extension and may then fall back to [`format_from_signature()`](http://readxl.tidyverse.org/reference/excel_format.html) to consult the [file's signature](https://en.wikipedia.org/wiki/List_of_file_signatures), a.k.a. its "magic number".
+
+`format_from_ext()` and, therefore, `excel_format()`, now recognize more file extensions in the xlsx family, namely `.xlsm`, `.xltx`, and `.xltm`, in addition to `.xlsx`.
+  
+```{r}
+library(readxl)
+library(tibble)
+
+tibble(
+  excel_paths  = c("a.xlsx", "b.xls", "c.xlsm", "d.xltx", "e.jpg", "f.csv", "g"),
+  format = excel_format(excel_paths, guess = FALSE) # consult extension **only**
+)
+```
+
+`format_from_signature()` and, therefore, `excel_format()` can determine the format of extensionless Excel files.
+
+```{r}
+guess_my_format <- c(
+  readxl_example("clippy.xls"),
+  readxl_example("datasets.xlsx"),
+  file.path(R.home("doc"), "html", "logo.jpg")
+)
+
+## copy and strip the extension
+filename <- tools::file_path_sans_ext(basename(guess_my_format))
+tmpdir <- tempdir()
+file.copy(from = guess_my_format, to = file.path(tmpdir, filename))
+
+tibble(
+  filename = filename,
+  format = excel_format(file.path(tmpdir, filename))
+)
+```
+
+## readxl in the real world
```

Comment:
This might be a nice recurring feature of our "package" blog posts, when relevant.

Reply:
looks good, thanks for the mention!